### PR TITLE
Issue/592/ccl v2.7

### PR DESCRIPTION
**Pull request by hsinfan1996**

### CODE REVIEW

File: clmm/theory/ccl.py
Comment (m-aguena) [inline clmm/theory/ccl.py:90]: Let me see if I got this correctly. We only need this line with `2.6.2dev7<CCL<2.7`, and with `CCL>2.7` `_set_concentration` does not work anymore?

**Comment by m-aguena** [comment]
@hsinfan1996 I will put this here to remember the suggestion on the co-work:

- [ ] Add verification of CCL version `XXX < v <YYY` where the current CLMM implementation works

**Comment by hsinfan1996** [comment]
> @hsinfan1996 I will put this here to remember the suggestion on the co-work:
> 
> * [x]  Add verification of CCL version `XXX < v <YYY` where the current CLMM implementation works

@m-aguena Done. v2.7.1dev10 uses v3 api and we should address it in another PR.

File: clmm/theory/ccl.py
Comment (hsinfan1996) [inline clmm/theory/ccl.py:107]: Here we do not support CCL >= 2.7.1.dev10 because we need change many places to make it work, not only this part. Do we want to write a comment about this fact explicitly?

File: clmm/theory/ccl.py
Comment (hsinfan1996) [inline clmm/theory/ccl.py:25]: With the release of 2.7.0, `'2.6.2dev7'` becomes `'2.7.1.dev1'`. This will make CLMM not support any release versions.

Comment (m-aguena) [reply clmm/theory/ccl.py:25]: The number written there was actually a placeholder, I was going to ask you the correct versions to fill vmin/vmax.
I moved that to ` clmm/theory/_ccl_supported_versions.py` now btw

File: clmm/theory/ccl.py
Comment (hsinfan1996) [inline clmm/theory/ccl.py:116]: Calling `UnlockInstance` and `_concentration_init` requires CCL between 2.7.1.dev1 and 2.7.1.dev9. Putting a if statement can extent the range to 2.6 to 2.7.1.dev9.

Comment (hsinfan1996) [reply clmm/theory/ccl.py:116]: Between 2.7.1.dev1 and 2.7.1.dev9 there is not conda installable versions, so that is why I hesitated to use only dev versions.

Comment (m-aguena) [reply clmm/theory/ccl.py:116]: please add it

**Review by m-aguena — APPROVED** [review]

**Comment by hsinfan1996** [comment]
@m-aguena Thank you.

**Comment by hsinfan1996** [comment]
@m-aguena pylint is not passing. Should I add module docstring or `# pylint: disable=missing-module-docstring` to `_ccl_supported_versions.py`?

**Comment by m-aguena** [comment]
I just added a docstring to the file, thanks

File: clmm/theory/ccl.py
Comment (hsinfan1996) [inline clmm/theory/ccl.py:119]: @m-aguena I am thinking if I should remove those two lines and make CLMM only support CCL 2.6 to 2.7, not the dev versions.

Comment (m-aguena) [reply clmm/theory/ccl.py:119]: Just leave it there for now. It will be removed when we update to CCL 3.0, right?

Comment (hsinfan1996) [reply clmm/theory/ccl.py:119]: For CCL v3, line 119 and 120 will become
```
with ccl.UnlockInstance(self.mdef):
      self.mdef.concentration = self.conc
```
and the if statement in line 118 will be removed

**Review by m-aguena — APPROVED** [review]

**Comment by hsinfan1996** [comment]
Is this one going to be merged soon? I kinda need this, though I can install from the branch.

**Comment by m-aguena** [comment]
It would be nice to have a second review. But you can marge it if no one replies by next week.

**Comment by hsinfan1996** [comment]
> It would be nice to have a second review. But you can marge it if no one replies by next week.

Yep. I see.

**Pull request merged**
